### ISSUE TITLE
Setting additionalProperties to false does not have any effect

### ISSUE THREAD

**Issue by connesc** [go-swagger/go-swagger#2035]
## Problem statement

I want to reject unknown fields provided in request body. According to [the FAQ](https://goswagger.io/faq/faq_model.html#extra-sections-in-post-body), it seems to be possible by simply adding `additionalProperties: false` to my definition.
However, this does not have any effect.

Is this a regression or just a misunderstanding?

## Swagger specification

```yaml
swagger: '2.0'
info:
  title: Example
  version: '0.1'
consumes:
- application/json
produces:
- application/json
paths:
  /test:
    post:
      operationId: test
      parameters:
      - name: body
        in: body
        schema:
          type: object
          properties:
            foo:
              type: boolean
          additionalProperties: false
      responses:
        200:
```

## Steps to reproduce

The following request should raise a 400 error, but does not:

```shell
curl -H 'Content-Type: application/json' -d '{"foo": true, "bar": false}' "http://127.0.0.1:$PORT/test"
```

## Environment
swagger version: v0.19.0
go version: go1.12.7 linux/amd64
OS: Linux

**Comment by fredbi** [comment]
https://goswagger.io/use/models/schemas.html#known-limitations-with-go-swagger-models

**Comment by connesc** [comment]
Thanks for the pointer, I didn't notice this doc before.

So at this point, I guess my best bet is to set `additionalProperties` to `true`, so that I can check that the `map[string]interface{}` is empty in my own code. However, it remains annoying because the Swagger doc will basically say the reverse of what it should.

Any idea for a better workaround?

**Comment by fredbi** [comment]
No, no other idea.

As explained in the link above, the trade-off in our implementation is currently to ignore additional properties when set to false, so this does not bear any validation semantics.
This is the same for `additionalItems`.

That being said, I am not against some opt-in behavior, e.g. `--strict-additional-properties` that would enforce validation (i.e. unwanted extra properties make the message invalid). You can add this kind of additional check in the custom unmarshaller template for schemas with additionalProperties.

**Comment by connesc** [comment]
@fredbi let me know if #2040 fits what you had in mind. At least, it perfectly solves my issue.
Thanks a lot for your help!

**Comment by connesc** [comment]
For future references, two alternatives are available:
- using `--strict-additional-properties` as implemented in #2040 (currently unsure whether it will be merged)
- using a custom `runtime.Consumer` as pointed out by @casualjim [here](https://github.com/go-swagger/go-swagger/pull/2040#issuecomment-520485853)
